pkg/fetch/fetcher.go

```diff
@@ -0,0 +1,21 @@
+package fetch
+
+import (
+	"context"
+
+	remoteasset "github.com/bazelbuild/remote-apis/build/bazel/remote/asset/v1"
+	"github.com/buildbarn/bb-remote-asset/pkg/qualifier"
+)
+
+// Fetcher is an abstraction around a Remote Asset API Fetch Server to allow for more consistent
+// Qualifier usage.
+type Fetcher interface {
```

Comment:
If I'm not mistaken this interface could have `FetchBlob` and `FetchDirectory` removed and replaced with just `remoteasset.FetchServer` like so:

```
type Fetcher interface {
  remoteasset.FetchServer
  CheckQualifiers(...)
}
```